core/common/src/main/java/alluxio/util/JvmPauseMonitor.java

```diff
@@ -0,0 +1,236 @@
+    GarbageCollectorMXBean newBean;
+    Set<String> nameSet = Sets.intersection(gcMXBeanMapBeforeSleep.keySet(),
+        gcMXBeanMapAfterSleep.keySet());
+    for (String name : nameSet) {
+      oldBean = gcMXBeanMapBeforeSleep.get(name);
+      newBean = gcMXBeanMapAfterSleep.get(name);
+      if (oldBean == null) {
+        beanDiffs.add("new GCBean created name= '" + newBean.getName() + " count="
+            + newBean.getCollectionCount() + " time=" + newBean.getCollectionTime() + "ms");
+      } else if (newBean == null) {
+        beanDiffs.add("old GCBean canceled name= '" + oldBean.getName() + " count="
+            + oldBean.getCollectionCount() + " time=" + oldBean.getCollectionTime() + "ms");
+      } else {
+        if (oldBean.getCollectionTime() != newBean.getCollectionTime()
+            || oldBean.getCollectionCount() != newBean.getCollectionCount()) {
+          beanDiffs.add("GC name= '" + newBean.getName() + " count="
+              + newBean.getCollectionCount() + " time=" + newBean.getCollectionTime() + "ms");
+        }
+      }
+    }
+    StringBuilder ret = new StringBuilder().append("JVM paused ").append(extraSleepTime)
+        .append("ms\n");
+    if (beanDiffs.isEmpty()) {
+      ret.append("No GCs detected ");
+    } else {
+      ret.append("GC list:\n" + Joiner.on("\n").join(beanDiffs));
+    }
+    ret.append("\n").append(getMemoryInfo());
+    return ret;
+  }
+
+  private Map<String, GarbageCollectorMXBean> getGarbageCollectorMXBeans() {
+    List<GarbageCollectorMXBean> gcBeanList = ManagementFactory.getGarbageCollectorMXBeans();
+    Map<String, GarbageCollectorMXBean> gcBeanMap = new HashMap();
+    for (GarbageCollectorMXBean gcBean : gcBeanList) {
+      gcBeanMap.put(gcBean.getName(), gcBean);
+    }
+    return gcBeanMap;
+  }
+
+  private class Monitor implements Runnable {
+    @Override
+    public void run() {
+      Stopwatch sw = new Stopwatch();
+      Map<String, GarbageCollectorMXBean> gcBeanMapBeforeSleep = getGarbageCollectorMXBeans();
+      while (true) {
+        sw.reset().start();
+        try {
+          Thread.sleep(mGcSleepIntervalMs);
+        } catch (InterruptedException ie) {
+          LOG.warn(ie.getStackTrace());
+          return;
+        }
+        long extraTime = sw.elapsed(TimeUnit.MILLISECONDS) - mGcSleepIntervalMs;
+        mTotalExtraTimeMs += extraTime;
+        Map<String, GarbageCollectorMXBean> gcBeanMapAfterSleep = getGarbageCollectorMXBeans();
+
+        if (extraTime > mWarnThresholdMs) {
+          mWarnTimeExceeded++;
+          LOG.warn(formatLogString(
+              extraTime, gcBeanMapBeforeSleep, gcBeanMapAfterSleep));
+        } else if (extraTime > mInfoThresholdMs) {
+          mInfoTimeExceeded++;
+          LOG.info(formatLogString(
+              extraTime, gcBeanMapBeforeSleep, gcBeanMapAfterSleep));
+        }
+        gcBeanMapBeforeSleep = gcBeanMapAfterSleep;
+      }
+    }
+  }
+
+  private class Daemon extends Thread {
```

Comment:
Since this is only used once, is it possible to make this part of the `Monitor` class?